### PR TITLE
Add fifo suffix, redrive policy, and additional outputs

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* [x] Allow periods in sqs fifo queue and sns topic if `fifo_topic` is true
* [x] Support `redrive_policy`
* [x] Added additional outputs

## why
* fifo queues and sns topic require `.fifo` in their names and the default regex in context is stripping out the period

## references
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sns_topic#name
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sqs_queue#name
* [x] Closes https://github.com/cloudposse/terraform-aws-sns-topic/issues/40
* [x] Closes https://github.com/cloudposse/terraform-aws-sns-topic/issues/42
* [x] Closes https://github.com/cloudposse/terraform-aws-sns-topic/issues/25

## test

```hcl
module "sns" {
  source = "git::https://github.com/cloudposse/terraform-aws-sns-topic.git?ref=sqs_queue_allow_periods"

  fifo_topic         = true
  sqs_dlq_enabled    = true
  fifo_queue_enabled = true
}
```

### CODE REVIEW

**Review by jamengual — DISMISSED** [review]

File: variables.tf [hunk 1]
```diff
@@ -86,8 +87,26 @@ variable "fifo_topic" {
   default     = false
 }
 
+variable "fifo_queue" {
```
Comment (aknysh) [inline variables.tf:90]:
```suggestion
variable "fifo_queue_enabled" {
```

Our convention/pattern

Comment (aknysh) [reply variables.tf:90]: the same for `fifo_topic_enabled`

**Review by aknysh — COMMENTED** [review]
please see comments

File: variables.tf (hunk 1 above)
Comment (nitrocode) [reply variables.tf]: Since fifo_topic is the same attribute as aws_sns_topic, shouldn't we keep that consistent ?

Comment (aknysh) [reply variables.tf]: let's keep it

File: outputs.tf [hunk 2]
```diff
@@ -1,9 +1,49 @@
...
+}
+
 output "aws_sns_topic_subscriptions" {
-  value       = aws_sns_topic_subscription.this
-  description = "SNS topic subscriptions"
+  value       = local.enabled ? aws_sns_topic_subscription.this : null
+  description = "SNS topic subscription"
+}
+
+output "sqs_queue_url" {
```
Comment (aknysh) [inline outputs.tf:31]:
```suggestion
output "dead_letter_queue_url" {
```

File: outputs.tf [hunk 3]
```diff
@@ -1,9 +1,49 @@
...
+  description = "The URL for the created Amazon SQS queue."
```
Comment (aknysh) [inline outputs.tf:32]:
```suggestion
  description = "Dead-letter queue URL"
```

File: outputs.tf [hunk 4]
```diff
@@ -1,9 +1,49 @@
 output "sns_topic" {
-  value       = aws_sns_topic.this
+  value       = local.enabled ? aws_sns_topic.this : null
```
Comment (aknysh) [inline outputs.tf:2]: `aws_sns_topic.this` will be a list with one item (or zero, if enabled=false).
Is it what we want?

File: outputs.tf [hunk 5]
```diff
@@ -1,9 +1,49 @@
...
+  value       = local.sqs_dlq_enabled ? aws_sqs_queue.dead_letter_queue[0].id : null
+}
+
+output "dead_letter_queue_name" {
+  description = "The name for the created dead letter queue."
+  value       = local.sqs_dlq_enabled ? local.sqs_queue_name : null
+}
+
+output "dead_letter_queue_arn" {
+  description = "The ARN of the dead letter queue"
```
Comment (aknysh) [inline outputs.tf:47]:
```suggestion
  description = "The ARN of the dead letter queue."
```

Let's be consistent, either use `.` at the end or don't :)

**Review by aknysh — COMMENTED** [review]
more comments :)

**Review by aknysh — APPROVED** [review]

**Pull request merged**